sqlmesh/core/config/loader.py

```diff
@@ -158,3 +175,10 @@ def load_config_from_env() -> t.Dict[str, t.Any]:
             target_dict[segments[-1]] = value
 
     return config_dict
+
+
+def convert_config_type(
+    config_obj: Config,
+    config_type: t.Type[C],
+) -> C:
+    return config_type.parse_obj(config_obj.dict())
```

Comment:
I've been recently working on something similar and I think you could reduce of the cost of this with this:
```
config_type(**{k: v for k, v in model})
```
That avoids having serde overhead of going to and from a dict by just iterating over the root level fields of the model.

Reply:
Oh I see good point